lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -196,6 +197,71 @@ def test_create_ccx(self):
         self.assertEqual(response.status_code, 200)
         self.assertTrue(re.search('id="ccx-schedule"', response.content))
 
+    @SharedModuleStoreTestCase.modifies_courseware
+    @patch('ccx.views.render_to_response', intercept_renderer)
+    @patch('ccx.views.TODAY')
+    def test_get_ccx_schedule(self, today):
+        """
+        Get CCX schedule and checks number of blocks in it
+        """
+        def assert_elements_in_schedule(
+                url,
+                n_chapters=2,
+                n_sequentials=4,
+                n_verticals=8
+            ):
+            """
+            Helper function to count visible elements in the schedule
+            """
+            response = self.client.get(url)
+            # the schedule contains chapters
+            chapters = json.loads(response.mako_context['schedule'])  # pylint: disable=no-member
+            sequentials = flatten(
+                [chapter.get('children', []) for chapter in chapters])
+            verticals = flatten(
+                [sequential.get('children', []) for sequential in sequentials])
+            self.assertEqual(n_chapters, len(chapters))
+            self.assertEqual(n_sequentials, len(sequentials))
+            self.assertEqual(n_verticals, len(verticals))
+            all_elements = chapters + sequentials + verticals
+            return [elem['location'] for elem in
+                    all_elements if 'location' in elem]
+
+
+        today.return_value = datetime.datetime(2014, 11, 25, tzinfo=pytz.UTC)
+        self.make_coach()
+        ccx = self.make_ccx()
+        url = reverse(
+            'ccx_coach_dashboard',
+            kwargs={
+                'course_id': CCXLocator.from_course_locator(
+                    self.course.id, ccx.id)
+            }
+        )
+        mstore = modulestore()
+        # all the elements are visible
+        assert_elements_in_schedule(url)
+        vertical = self.verticals[0]
+        vertical.visible_to_staff_only = True
+        mstore.update_item(vertical, self.coach.id)
+        locations = assert_elements_in_schedule(url, n_verticals=7)
+        self.assertNotIn(unicode(vertical.location), locations)
+        # hide a sequential
+        sequential = self.sequentials[0]
+        sequential.visible_to_staff_only = True
+        mstore.update_item(sequential, self.coach.id)
+        locations = assert_elements_in_schedule(
+            url, n_sequentials=3, n_verticals=6)
```

Comment:
I bet this can fit on one line, but if you're gonna break lines please see https://github.com/edx/edx-platform/wiki/Python-Guidelines#breaking-long-lines - partiularly closing paren should go on next line (throughout PR)